src/openms/include/OpenMS/ANALYSIS/ID/ConsensusMapMergerAlgorithm.h

```diff
@@ -0,0 +1,151 @@
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julianus Pfeuffer $
+// $Authors: Julianus Pfeuffer $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/CONCEPT/Exception.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/KERNEL/ConsensusMap.h>
+#include <OpenMS/METADATA/ExperimentalDesign.h>
+
+#include <unordered_set>
+
+namespace OpenMS
+{
+  /**
+   * @brief Merges identification data in ConsensusMaps
+   * Has some stuff in common with IDMergerAlgorithm and therefore could be merged
+   * but you can save some overhead by only going through the CMap once. Therefore
+   * the extra class.
+   */
+  class OPENMS_DLLAPI ConsensusMapMergerAlgorithm:
+    public DefaultParamHandler,
+    public ProgressLogger
+  {
+  public:
+    ConsensusMapMergerAlgorithm ();
+    /// Takes a cmap with one IDRun per column and merges them to one proteinIDRun per Condition
+    /// while reassociating the PeptideEvidences
+    /// Constructs the mapping based on the exp. design and then uses mergeProteinIDRuns
+    /// @throws MissingInformationException for e.g. missing map_indices in PeptideIDs
+    void mergeProteinsAcrossFractionsAndReplicates(ConsensusMap& cmap, const ExperimentalDesign& exp_design) const;
+
+    /// Similar to above, merges every ID Run into one big run. Proteins get only inserted once but Peptides stay unfiltered
+    /// i.e. might occur in several PeptideIdentifications afterwards
+    /// @throws MissingInformationException for e.g. missing map_indices in PeptideIDs
+    void mergeAllIDRuns(ConsensusMap& cmap) const;
```

Comment:
same here. what happens to groups?

Reply:
Of course groups don't make sense anymore and are removed. added note.